translate/pb/pb_util.go

```diff
@@ -0,0 +1,163 @@
+package pb
+
+import (
+	"fmt"
+
+	"github.com/arr-ai/arrai/rel"
+	"github.com/arr-ai/arrai/tools"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+// decodeFileDescriptor parses protobuf definition file and decodes to file descriptor
+func decodeFileDescriptor(definition []byte) (protoreflect.FileDescriptor, error) {
+	fdSet := new(descriptorpb.FileDescriptorSet)
+	if err := proto.Unmarshal(definition, fdSet); err != nil {
+		return nil, err
+	}
+
+	// We know protoc was invoked with a single .proto file only
+	file1 := fdSet.GetFile()[0]
+
+	// Initialize the File descriptor object
+	fd1, err := protodesc.NewFile(file1, protoregistry.GlobalFiles)
+	if err != nil {
+		return nil, err
+	}
+
+	return fd1, nil
+}
+
+// WalkThroughMessageToBuildValue walks through protobuf message and build a value whose type is rel.Value
```

Comment:
Nit: Naming: while you do use this function to walk through the message, it's actually more general than that. Looking at the signature, it's converting a `protoreflect.Value` into a `rel.Value`. In the case where the input is a Message, it will walk through it, but in other cases it doesn't.

Maybe something like `ConvertProtoValueToSysl`?